src/sentry/incidents/logic.py

```diff
@@ -812,17 +818,17 @@ def update_alert_rule(
             comparison_delta = int(timedelta(minutes=comparison_delta).total_seconds())
 
         updated_fields["comparison_delta"] = comparison_delta
-
-    # if we modified the comparison_delta or the time_window, we should update the resolution accordingly
-    if "comparison_delta" in updated_fields or "time_window" in updated_query_fields:
-        if comparison_delta is not None:
-            detection_type = AlertRuleDetectionType.PERCENT
-        else:
-            if seasonality:
-                detection_type = AlertRuleDetectionType.DYNAMIC
+    if detection_type is not None:
+        updated_fields["detection_type"] = detection_type
```

Comment:
should users be able to swap between detection types?
we should still clear comparison delta if the type is _not_ percent right?

Reply:
`we should still clear comparison delta if the type is not percent right?`
I believe we do this ^^
Users are able to swap detection types.